src/client/components/UserPage/UserLinkTable/ToolBar/FilterSortPanel/index.jsx

```diff
@@ -0,0 +1,116 @@
+import SortPanel from './SortPanel'
+import userActions from '../../../../../actions/user'
+import FilterPanel from './FilterPanel'
+import FilterSortPanelFooter from './FilterSortPanelFooter'
+
+import closeIcon from '~/assets/icons/close-icon.svg'
+import useStyles from './styles'
+
+const mapDispatchToProps = (dispatch) => ({
+  updateSortAndFilter: (title, direction, state, isFile) => {
+    dispatch(
+      userActions.setUrlTableConfig({
+        orderBy: title,
+        sortDirection: direction,
+        filter: {
+          state,
+          isFile,
+        },
+      }),
+    )
+    dispatch(userActions.getUrlsForUser())
+  },
+})
+
+const FilterSortPanel = ({
+  isOpen,
+  onClose,
+  tableConfig,
+  updateSortAndFilter,
+}) => {
+  const classes = useStyles()
+  const [orderBy, setOrderBy] = useState(tableConfig.orderBy)
+  const [isIncludeFiles, setIsIncludeFiles] = useState(
+    tableConfig.filter.isFile === true,
+  )
+  const [isIncludeLinks, setIsIncludeLinks] = useState(
+    tableConfig.filter.isFile === false,
+  )
+  const [isIncludeActive, setIsIncludeActive] = useState(
+    tableConfig.filter.state === 'ACTIVE',
+  )
+  const [isIncludeInactive, setIsIncludeInactive] = useState(
+    tableConfig.filter.state === 'INACTIVE',
+  )
+  const filterConfig = {
+    isIncludeFiles,
+    isIncludeLinks,
+    isIncludeActive,
+    isIncludeInactive,
+    setIsIncludeFiles,
+    setIsIncludeLinks,
+    setIsIncludeActive,
+    setIsIncludeInactive,
+  }
+  const changeSortAndFilterHandler = () => {
+    let isFile
+    let state
+    if (isIncludeLinks !== isIncludeFiles) {
+      if (isIncludeLinks) {
+        isFile = false
+      } else {
+        isFile = true
+      }
+    }
+    if (isIncludeActive !== isIncludeInactive) {
+      if (isIncludeActive) {
+        state = 'ACTIVE'
+      } else {
+        state = 'INACTIVE'
+      }
+    }
+    updateSortAndFilter(orderBy, 'desc', state, isFile)
+    onClose()
+  }
+  const reset = () => {
+    setIsIncludeFiles(false)
+    setIsIncludeLinks(false)
+    setIsIncludeActive(false)
+    setIsIncludeInactive(false)
+    setOrderBy('updatedAt')
+  }
+
+  return (
+    <Collapse
+      in={isOpen}
+      className={classes.collapse}
+      classes={{ wrapper: classes.collapseWrapper }}
+      timeout={100}
+    >
+      <Paper className={classes.root}>
+        <IconButton className={classes.closeIcon} onClick={onClose}>
+          <img src={closeIcon} alt="Close" draggable={false} />
+        </IconButton>
+        <Grid
+          container
+          style={{
+            paddingTop: '32px',
+            paddingBottom: '48px',
+          }}
+        >
+          <SortPanel onChoose={setOrderBy} currentlyChosen={orderBy} />
+          <FilterPanel {...filterConfig} />
+          <FilterSortPanelFooter
+            onApply={changeSortAndFilterHandler}
+            onReset={reset}
+          />
+        </Grid>
+      </Paper>
+    </Collapse>
```

Comment:
Can I check why are we not using the MuiMenu for this?

Reply:
Do you think we also split styles for widgets within component directories to have their own dedicated styles file too?

Reply:
And rather than using box-icons, I believe we can download the assets and put them into our project directly.

Reply:
Mui menu doesn't allow us to keep the dimensions equal to the parent. 

We could do that but I noticed that quite a few styles are shared so I opted to share it instead.

Reply:
Should we keep to using box-icons since we can easily manipulate the colour without duplication the asset?

Reply:
For the newer components, I have been using local assets in place of box-icons. It is easily better to do it progressively

Reply:
> Mui menu doesn't allow us to keep the dimensions equal to the parent.
> 
> We could do that but I noticed that quite a few styles are shared so I opted to share it instead.

I am worried about the trade-off, since we are using an absolute offset for the menu now

Reply:
The offset is in relation to the top of the search input so its simple to manage. It's way more complex to ensure that the two widths are equal without it as menu does not provide this form of functionality, especially if the width changes when resizing the page

Reply:
I believe the width of the search input does not change with varying desktop view width. Even if it is so, we should be able to override the paper property in Menu to fix the menu width. And to better connect the components, we can also abstract the width of the search input into a hook.